From 3c4bf2609ee8196d47717b2e133189ba3da4657f Mon Sep 17 00:00:00 2001
From: Yuri Lin <yurilin@google.com>
Date: Wed, 12 Oct 2022 14:27:46 +0000
Subject: [PATCH] [DO NOT MERGE] Fix conditionId string trimming in
 AutomaticZenRule

This change only applies to S branches and earlier.

Bug: 253085433
Bug: 242703460
Bug: 242703505
Bug: 242703780
Bug: 242704043
Bug: 243794204
Test: AutomaticZenRuleTest
Change-Id: Iae423d93b777df8946ecf1c3baf640fcf74990ec
Merged-In: Iae423d93b777df8946ecf1c3baf640fcf74990ec
(cherry picked from commit 83d23fb275d2bcfb090a9c6efd6c71a5f519372f)
Merged-In: Iae423d93b777df8946ecf1c3baf640fcf74990ec
---
 core/java/android/app/AutomaticZenRule.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/app/AutomaticZenRule.java b/core/java/android/app/AutomaticZenRule.java
index 37b336382769..9a92515ee794 100644
--- a/core/java/android/app/AutomaticZenRule.java
+++ b/core/java/android/app/AutomaticZenRule.java
@@ -125,7 +125,7 @@ public final class AutomaticZenRule implements Parcelable {
             name = getTrimmedString(source.readString());
         }
         interruptionFilter = source.readInt();
-        conditionId = source.readParcelable(null);
+        conditionId = getTrimmedUri(source.readParcelable(null));
         owner = getTrimmedComponentName(source.readParcelable(null));
         configurationActivity = getTrimmedComponentName(source.readParcelable(null));
         creationTime = source.readLong();
-- 
2.39.0.rc1.256.g54fd8350bd-goog

